File: lesunny/055_제일작은수제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        int[] answer;
        int length = arr.length;
        
        if(length == 1){
            answer = new int[1];
            answer[0] = -1;
        }else{
            answer = new int[length-1];
            
            int min = 100;
            for(int i=0; i<length; i++){
                min = Math.min(min, arr[i]);
            }
            int idx=0;
            for(int i=0; i<length; i++){
                if(min != arr[i])
                    answer[idx++] = arr[i];
            }
        }
        return answer;
    }
}

File: jiwon/055 제일 작은 수 제거하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int min = 100000;
    int idx = 0;
    
    for(int i=0;i<arr.size();i++){
        if(arr.size() == 1 || arr.size() == 0) {
            answer.push_back(-1);
            return answer;
        }
        
        if(min > arr[i]) {
            min = arr[i];
            idx = i;
        }
    }
    
    arr.erase(arr.begin() + idx);
    
    return arr;
}
